src/Fetch.js

```diff
@@ -554,7 +570,18 @@ function emscripten_start_fetch(fetch, successcb, errorcb, progresscb) {
   }
   return fetch;
 #else // !FETCH_SUPPORT_INDEXEDDB
-  __emscripten_fetch_xhr(fetch, reportSuccess, reportError, reportProgress);
+  __emscripten_fetch_xhr(fetch, reportSuccess, reportError, reportProgress, reportReadyStateChange);
   return fetch;
 #endif // ~FETCH_SUPPORT_INDEXEDDB
 }
+
+function _emscripten_fetch_get_response_headers_length(id) {
+    return lengthBytesUTF8(Fetch.xhrs[id-1].getAllResponseHeaders()) + 1;
+}
+
+function _emscripten_fetch_get_response_headers(id, dst, dstSizeBytes) {
+    var responseHeaders = Fetch.xhrs[id-1].getAllResponseHeaders();
```

Comment:
Ohh, needing to do `id-1` here is actually a bug! I had not intended this. The `id` field should immediately be indexable to `xhrs` array. However, let's leave this change as a followup, I see there are other issues in play here that need remedying.

Reply:
Sure 👍